src/MetaCompilation.Analyzers/Core/DiagnosticAnalyzer.cs

```diff
@@ -2603,7 +2603,7 @@ private bool CheckMethods(InvocationExpressionSyntax invocationExpression, Compi
                         ReportDiagnostic(context, IncorrectAnalysisReturnTypeRule, analysisMethodSyntax.Identifier.GetLocation(), analysisMethodSyntax.Identifier.ValueText);
                         return false;
                     }
-                    else if (analysisMethod.Parameters.Count() != 1 || analysisMethod.Parameters.First().Type != context.Compilation.GetTypeByMetadataName("Microsoft.CodeAnalysis.Diagnostics.SyntaxNodeAnalysisContext"))
+                    else if (analysisMethod.Parameters.Count() != 1 || !Equals(analysisMethod.Parameters.First().Type, context.Compilation.GetTypeByMetadataName("Microsoft.CodeAnalysis.Diagnostics.SyntaxNodeAnalysisContext")))
```

Comment:
I presume this analyzer is enabled by default (and also other analyzers you recently added)? If not, please ensure we turn them on for the repo by updating https://github.com/dotnet/roslyn-analyzers/blob/master/eng/Analyzers_ShippingRules.ruleset.

Please also update our Microsoft.CodeAnalysis.Analyzers [version](https://github.com/dotnet/roslyn-analyzers/blob/master/eng/Versions.props#L24) to the latest myget package so we enforce it on build. Thanks!

Reply:
Yep, I'm working on that in a separate PR.